galaxy/main/management/commands/recompute_quality.py

```diff
@@ -0,0 +1,163 @@
+"""Recomputes the Quality Score
+
+Recomputes the quality score for a collection and all its content.
+
+Recomputes quality score for collections which already have a
+quality score. Does not reimport or relint - uses existing linter violations.
+
+Run from inside a container, examples:
+
+/var/lib/galaxy/venv/bin/python /galaxy/manage.py recompute_quality --help
+
+/var/lib/galaxy/venv/bin/django-admin recompute_quality --help
+"""
+
+from django.core.management.base import BaseCommand
+from galaxy.main.models import Repository, ImportTask
+from galaxy.worker.tasks import _update_quality_score
+from datetime import datetime
+import pytz
+import sys
+import os
+
+
+class NoStdout():
+    def __enter__(self):
+        self.stdout = sys.stdout
+        self.file = open(os.devnull, 'w')
+        sys.stdout = self.file
+
+    def __exit__(self, *args):
+        sys.stdout = self.stdout
+        self.file.close()
+
+
+class Command(BaseCommand):
+    help = ('Recomputes quality score for collections which already have a '
+            'quality score. Does not reimport or relint - '
+            'uses existing linter violations.')
+    repos = []
+    scored_all, scored_before, scored_after = None, None, None
+
+    def add_arguments(self, parser):
+        parser.add_argument(
+            '--count', action='store_true',
+            help=('Get count of collections with a non-null quality '
+                  'score based on --scored-* option.')
+        )
+        parser.add_argument(
+            '--recompute', action='store_true',
+            help=('Recomputes collections with scores based '
+                  'on --scored-* option. Uses existing linter violations '
+                  'and latest severities, does not relint or reimport')
+        )
+        parser.add_argument(
+            '--scored-all', action='store_true',
+            help=('Looks at collections scored at any time')
+        )
+        parser.add_argument(
+            '--scored-before', nargs=1, type=str,
+            help=('Looks at collections scored before given datetime, '
+                  'use format 2018-10-29-23:30:00')
+        )
+        parser.add_argument(
+            '--scored-after', nargs=1, type=str,
+            help=('Looks at collections scored after given datetime, '
+                  'use format 2018-10-29-23:30:00')
+        )
+
+    def handle(self, *args, **kwargs):
+        try:
+            self.validate_args(kwargs)
```

Comment:
Could validate_args be replaced with a argparse mutually exclusive groups, ala https://docs.python.org/3/library/argparse.html#mutual-exclusion ?

Reply:
Yes that worked nicely